test/examples/ucp_hello_world.c

```diff
@@ -235,34 +193,65 @@ static int run_ucx_client(ucp_worker_h *ucp_worker)
     free (msg);
 
     /* Receive test string from server */
-    for ( ; ; ucp_worker_progress(*ucp_worker)) {
-        msg_tag = ucp_tag_probe_nb(*ucp_worker, tag, (ucp_tag_t)-1, 1,
-                                   &info_tag);
-        if (msg_tag == NULL) {
-            continue;
+    if (ucp_test_mode == TEST_MODE_POLL || ucp_test_mode == TEST_MODE_POLL_GET) {
+        if (ucp_test_mode == TEST_MODE_POLL) {
+            /* Polling incoming events*/
+            status = ucp_worker_wait(ucp_worker);
+            if (status != UCS_OK) {
+                goto err_ep;
+            }
+        } else if (ucp_test_mode == TEST_MODE_POLL_GET) {
+            status = test_poll_wait(ucp_worker);
+            if (status != UCS_OK) {
+                goto err_ep;
+            }
+        } else {
+            fprintf(stderr, "ERROR: Wrong test mode selected in "
+                    "run_ucx_client %d\n", ucp_test_mode);
+            goto err_ep;
+        }
+        msg_len = sizeof(*msg) + strlen(test_str) + 1;
+        msg = malloc(msg_len);
+        if (!msg) {
+            fprintf(stderr, "unable to allocate memory in polling branch\n");
+            goto err_ep;
+        }
+        request = ucp_tag_recv_nb(ucp_worker, msg, msg_len,
+                ucp_dt_make_contig(1), tag,
+                tag_mask, recv_handle);
+    } else if (ucp_test_mode == TEST_MODE_PROBE) {
+        /* Probing incoming events*/
+        for ( ; ; ucp_worker_progress(ucp_worker)) {
+            msg_tag = ucp_tag_probe_nb(ucp_worker, tag, tag_mask, 1,
```

Comment:
we should have "probe" in POLL/GET, in order to demonstrate how to use it to get the message size. so we could unite most of the code in these flows.